gematria/granite/python/graph_builder_model_base.py

```diff
@@ -187,11 +190,13 @@ def _create_tf_graph(self) -> None:
   # @Override
   def _create_readout_network_resources(self) -> None:
     super()._create_readout_network_resources()
-    self._instruction_node_mask = tf.placeholder(
-        dtype=tf.dtypes.bool,
-        shape=(None,),
-        name=GraphBuilderModelBase.INSTRUCTION_NODE_MASK_TENSOR_NAME,
-    )
+    # May already be created by subclass.
+    if self._instruction_node_mask is None:
+      self._instruction_node_mask = tf.placeholder(
+          dtype=tf.dtypes.bool,
+          shape=(None,),
+          name=GraphBuilderModelBase.INSTRUCTION_NODE_MASK_TENSOR_NAME,
+      )
```

Comment:
Under what circumstances would it be created by a subclass? (and does the subclass need to do that?)

Reply:
It would always be created by `TokenGraphBuilderModel`s - to figure out what nodes are instructions and to update their embeddings with annotations. We don't really need this since we won't be making instances of `GraphBuilderModelBase`s, but I left it in to ensure it is there in case there is ever another subclass that doesn't create it.

Another option is to create it within `GraphBuilderModelBase` earlier, so that it is available to `TokenGraphBuilderModel`s when required.

Reply:
This will be removed since moving the creation of `_instruction_node_mask` should be done alongside the creation of annotation-related tensors which is moving to `GraphBuilderModelBase` as suggested in another comment.